src/client/runner/utils/promise-each.js

```diff
@@ -0,0 +1,23 @@
+import hammerhead from '../deps/hammerhead';
+
+var Promise = hammerhead.Promise;
+
+export default function (items, iterator) {
+    var index = 0;
```

Comment:
If I have understood right, purpose of the function is to pass each element of `items` to `iterator` and chain returned promises? E.g
```
items = [i1,i2,i3]

//after calling promise-each

iterator(i1)
  .then(() => iterator(i2))
  .then(() => iterator(i3))
```
So what's about 
```
export default function (items, iterator) {
    return items
        	.reduce((promise, item) => promise.then(() => iterator(item)), Promise.resolve());        
}
```
IMHO it looks cleaner.

It uses fact that if you return a Promise in `then` handler, it will be resolved before calling next `then` handler in Promise chain ([explanation](https://learn.javascript.ru/promise#цепочки-промисов)). 
Also look at [Array.prototype.reduce](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce).